src/screens/Settings/HSwitchProfile/TwoFaSettings/ResetTotp.res

```diff
@@ -0,0 +1,239 @@
+      }
+    }
+  }
+
+  let verifyTOTP = async (~fromModal, ~methodType, ~otp) => {
+    try {
+      setButtonState(_ => Button.Loading)
+      if otpInModal->String.length > 0 || otp->String.length > 0 {
+        let body = [("totp", otp->JSON.Encode.string)]->getJsonFromArrayOfJson
+
+        let _ = await verifyTotpLogic(body, methodType)
+        if fromModal {
+          setShowVerifyModal(_ => false)
+          generateNewSecret()->ignore
+        } else {
+          showToast(~message="Successfully reset the totp !", ~toastType=ToastSuccess)
+          RescriptReactRouter.push(GlobalVars.appendDashboardPath(~url="/account-settings/profile"))
+        }
+        setOtp(_ => "")
+        setOtpInModal(_ => "")
+      } else {
+        showToast(~message="OTP field cannot be empty!", ~toastType=ToastError)
+      }
+      setButtonState(_ => Button.Normal)
+    } catch {
+    | Exn.Error(e) => {
+        let err = Exn.message(e)->Option.getOr("Verification Failed")
+        let errorMessage = err->safeParse->getDictFromJsonObject->getString("message", "")
+        let errorCode = err->safeParse->getDictFromJsonObject->getString("code", "")
+        if errorCode->CommonAuthUtils.errorSubCodeMapper === UR_42 {
+          setTotpSecret(_ => RegenerateQR)
+        }
+        setOtpInModal(_ => "")
+        setOtp(_ => "")
+        setErrorMessage(_ => errorMessage)
+        setButtonState(_ => Button.Normal)
+      }
+    }
+  }
+
+  let verifyRecoveryCode = async () => {
+    try {
+      setButtonState(_ => Button.Loading)
+      if recoveryCode->String.length > 0 {
+        let body = [("recovery_code", recoveryCode->JSON.Encode.string)]->getJsonFromArrayOfJson
+        let _ = await verifyRecoveryCodeLogic(body)
+        setShowVerifyModal(_ => false)
+      } else {
+        showToast(~message="Recovery code cannot be empty!", ~toastType=ToastError)
+      }
+      setRecoveryCode(_ => "")
+      setButtonState(_ => Button.Normal)
+    } catch {
+    | Exn.Error(e) => {
+        let err = Exn.message(e)->Option.getOr("Verification Failed")
+        let errorMessage = err->safeParse->getDictFromJsonObject->getString("message", "")
+        setRecoveryCode(_ => "")
+        setErrorMessage(_ => errorMessage)
+        setButtonState(_ => Button.Normal)
+      }
+    }
+  }
+
+  let handle2FaVerify = () => {
+    switch twoFaState {
+    | Totp => verifyTOTP(~fromModal=true, ~methodType=Post, ~otp={otpInModal})
+    | RecoveryCode => verifyRecoveryCode()
+    }
+  }
+
+  React.useEffect(() => {
+    if checkStatusResponse.totp || checkStatusResponse.recovery_code {
+      generateNewSecret()->ignore
+    } else {
+      setShowVerifyModal(_ => true)
+    }
+    None
+  }, [])
+
+  let handleKeyUp = ev => {
+    open ReactEvent.Keyboard
+    let key = ev->key
+    let keyCode = ev->keyCode
+
+    if key === "Enter" || keyCode === 13 {
+      handle2FaVerify()->ignore
+    }
+  }
```

Comment:
only keyCode is sufficent to check I feel as keyCode is unique
```suggestion
  let handleKeyUp = ev => {
    let keyCode = ev->ReactEvent.Keyboard.keyCode
    if keyCode === 13 {
      handle2FaVerify()->ignore
    }
  }
```

Reply:
By using both checks (key === 'Enter' || keyCode === 13), we ensure that your application works across all browsers, and for older and newer versions